pkg/resource/internal/compare/compare.go

```diff
@@ -0,0 +1,108 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package compare implements term operation helpers.
+package compare
+
+import (
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+// GetNumbers returns numbers parsed from left and right params, if any of string to number conversion fails returns false as the 3rd arg.
+func GetNumbers(left, right string) (int64, int64, bool) {
+	numLeft, ok := parseValue(left)
+	if !ok {
+		return 0, 0, false
+	}
+
+	numRight, ok := parseValue(right)
+	if !ok {
+		return 0, 0, false
+	}
+
+	return numLeft, numRight, true
+}
+
+func parseValue(value string) (int64, bool) {
+	value = strings.TrimSpace(value)
+
+	var (
+		digits string
+		rest   string
+	)
+
+	for _, letter := range value {
+		if rest == "" && unicode.IsDigit(letter) || string(letter) == "." {
+			digits += string(letter)
+
+			continue
+		}
+
+		rest += string(letter)
+	}
+
+	if len(digits) == 0 {
+		return 0, false
+	}
+
+	res, err := strconv.ParseInt(digits, 10, 64)
+	if err != nil {
+		return 0, false
+	}
+
+	return res * getMultiplier(rest), true
```

Comment:
not clear if we allow invalid `multiplier` here? should it be documented? 

e.g. `1x` parses as `1` as it seems

Reply:
I guess we can just disable invalid `multipliers` here and it will also solve this `.` issue as well.